vault/resource_database_secret_backend_connection_test.go

```diff
@@ -139,6 +139,41 @@ func TestAccDatabaseSecretBackendConnection_cassandraProtocol(t *testing.T) {
 	})
 }
 
+func TestAccDatabaseSecretBackendConnection_mongodbatlas(t *testing.T) {
+	public_key := os.Getenv("MONGODB_ATLAS_PUBLIC_KEY")
```

Comment:
Changing this will make the test the odd one out, as all the tests in this file follow the pattern "if environment variable X is not set, skip test". It does this for every supported backend connection. 

Ofcourse I have no problem making the changes as per your request but I would like to get confirmation this is indeed how it's supposed to be :)

Reply:
Yeah fair enough - I have asked around internally and no one knows why it's the way it is. But that's a separate issue to this PR. I'm fine with merging this in its consistent state, and the environment variable weirdness will get addressed separately if necessary.